R/summariseClinicalRecords.R

```diff
@@ -47,80 +47,77 @@
+                           cli::cli_warn(paste0(x, " omop table is empty. Returning an empty summarised omop table."))
+                           return(omopgenerics::emptySummarisedResult())
+                         }
+                         summariseClinicalRecord(x,
+                                                 cdm = cdm,
+                                                 recordsPerPerson = recordsPerPerson,
+                                                 inObservation = inObservation,
+                                                 standardConcept = standardConcept,
+                                                 sourceVocabulary = sourceVocabulary,
+                                                 domainId = domainId,
+                                                 typeConcept = typeConcept)
+                       }
+                       ) |>
+    dplyr::bind_rows()
 
   return(result)
 }
 
 #' @noRd
-summariseClinicalRecord <- function(cdm,
-                                    tableName,
-                                    recordsPerPerson = c("mean", "sd", "median", "q25", "q75", "min", "max"),
-                                    inObservation = TRUE,
-                                    standardConcept = TRUE,
-                                    sourceVocabulary = FALSE,
-                                    domainId = TRUE,
-                                    typeConcept = TRUE) {
-
-  assertClass(cdm[[tableName]], "omop_table")
-  omopTable <- cdm[[tableName]] |> dplyr::ungroup()
-
-  if ("observation_period" == tableName) {
+summariseClinicalRecord <- function(omopTableName, cdm, recordsPerPerson,
+                                    inObservation, standardConcept,
+                                    sourceVocabulary, domainId, typeConcept,
+                                    call = parent.frame(3)) {
+
+  omopgenerics::assertClass(cdm[[omopTableName]], "omop_table", call = call)
+  omopTable <- cdm[[omopTableName]] |> dplyr::ungroup()
+
+  if ("observation_period" == omopTableName) {
     if(standardConcept){
       if(!missing(standardConcept)){
-        cli::cli_warn("standardConcept turned to FALSE, as observation_period is in tableNames")
+        cli::cli_warn("standardConcept turned to FALSE, as observation_period is in omopTableName", call = call)
       }
       standardConcept <- FALSE
     }
     if(sourceVocabulary){
       if(!missing(sourceVocabulary)){
-        cli::cli_warn("sourceVocabulary turned to FALSE, as observation_period is in tableNames")
+        cli::cli_warn("sourceVocabulary turned to FALSE, as observation_period is in omopTableName", call = call)
       }
       sourceVocabulary <- FALSE
     }
     if(domainId){
       if(!missing(domainId)){
-        cli::cli_warn("domainId turned to FALSE, as observation_period is in tableNames")
+        cli::cli_warn("domainId turned to FALSE, as observation_period is in omopTableName", call = call)
```

Comment:
call not used in warnings -> either use cli::cli_alert()/cli::cli_inform() and they will be messages which is fine